src/Network/OAuth/OAuth2/HttpClient.hs

```diff
@@ -160,6 +173,29 @@ parseResponseJSON (Right b) = case decode b of
                             Nothing -> Left ("hoauth2.HttpClient.parseResponseJSON/Could not decode JSON: " `BSL.append` b)
                             Just x -> Right x
 
+-- | Parses a @OAuth2Result BSL.ByteString@ that contains not JSON but a Query String
+parseResponseUrl :: FromJSON a
```

Comment:
nit: maybe name it `parseResponseString`. `parseResponseUrl` is a little confusing to me.